django/contrib/auth/forms.py

```diff
@@ -194,6 +238,11 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         password = self.fields.get("password")
         if password:
+            if self.instance and not self.instance.has_usable_password():
+                password.help_text = _(
+                    "Enable password-based authentication for this user by setting "
+                    'a password using <a href="{}">this form</a>.'
```

Comment:
https://design-system.service.gov.uk/components/text-input/#hint-text

What do you think about avoiding links in help text?

> Do not include links within hint text. While screen readers will read out the link text when describing the field, they will not tell users that the text is a link.

Reply:
@smithdc1 I personally think this should be a **button** but did not want to interfere with #17419.

Reply:
@smithdc1 I kind of agree with you but if we remove the link from the help text, how/where can we expose the change password functionality? Any ideas or suggestions? (in any case, I would do that in a separated ticket/PR)

Reply:
@nessita I'd make it a button, just blow the help text. It'll be a simple string change that can be easily added to this PR. It would look like this: 
<img width="694" alt="image" src="https://github.com/django/django/assets/16904477/eeee524b-3845-411a-a804-7217d8778e62">
I can add this here or create a separate mini-PR. What's your preference?

Reply:
I think this would be an amazing change. Whether it’s a separate PR or not – not my call.

Reply:
I think this would be an amazing change. Whether as a separate PR or not – not my call.

Reply:
@smithdc1 @thibaudcolas I created this PR #17489 to introduce the set/change password button.